rts/motoko-rts/src/principal_id.rs

```diff
@@ -5,15 +5,15 @@ use crate::mem_utils::memcpy_bytes;
 use crate::memory::{alloc_blob, Memory};
 use crate::rts_trap_with;
 use crate::text::{blob_compare, blob_of_text};
-use crate::types::{Bytes, Value, TAG_BLOB};
+use crate::types::{Bytes, Value, TAG_BLOB_B, TAG_BLOB_T};
 
 use motoko_rts_macros::ic_mem_fn;
 
 // CRC32 for blobs. Loosely based on https://rosettacode.org/wiki/CRC-32#Implementation_2
 
 #[no_mangle]
 pub unsafe extern "C" fn compute_crc32(blob: Value) -> u32 {
-    if blob.tag() != TAG_BLOB {
+    if !blob.is_blob() {
```

Comment:
Would this be a case where the blob is a principal? The check could then be constrained to `TAG_BLOB_P`.

Reply:
**Errand**: Maybe not. Below it seems to be a `TAG_BLOB_B` that is passed to this function. It is hard to follow...